docs/CONTRIBUTING.md

```diff
@@ -199,3 +199,51 @@ one or more reviewers making new comments every time, then waiting for an
 updated change before reviewing again. All contributors, including those from
 maintainers, are subject to the same review cycle; this process is not meant to
 be applied selectively or to discourage anyone from contributing.
+
+## Releases
+
+Coder releases are automated via GitHub Actions. Specifically, the
+[`release.yaml`](../.github/workflows/release.yaml) workflow. They are created
+based on the current [`main`](https://github.com/coder/coder/tree/main) branch.
+
+The release notes for a release are automatically generated from commit titles
+and metadata from PRs that are merged into `main`.
+
+### Creating a release (script)
+
+The preferred way to create a release is to use [`./scripts/release.sh`](../scripts/release.sh). This script will show a preview of the release that _would_ be created and if you choose to continue, automatically run and watch the release workflow for you.
+
+See `./scripts/release.sh --help` for more information.
+
+### Creating a release (workflow dispatch)
+
+Alternatively, the workflow can be dispatched manually from [Actions: Release](https://github.com/coder/coder/actions/workflows/release.yaml). Simply press "Run workflow" and set your preferred inputs.
+
+### Commit messages
+
+Commit messages should follow the [Conventional Commits 1.0.0](https://www.conventionalcommits.org/en/v1.0.0/) specification.
+
+Allowed commit types (`feat`, `fix`, etc.) are listed in [conventional-commit-types](https://github.com/commitizen/conventional-commit-types/blob/c3a9be4c73e47f2e8197de775f41d981701407fb/index.json). Note that these types are also used to automatically sort and organize the release notes.
+
+A good commit message title uses the imperative, present tense and is ~50
+characters long (no more than 72).
+
+Examples:
+
+- Good: `feat(api): Add feature X`
+- Bad: `feat(api): Added feature X` (past tense)
+
+A good rule of thumb for writing good commit messages is to recite: [If applied, this commit will ...](https://reflectoring.io/meaningful-commit-messages/).
```

Comment:
When I merge PRs, I make sure that the first line of commit message is aligned with the convention. Should I also pay attention to other lines? For instance:

```
commit d9436fab69203740a25c29f2d27146bb04bc14c3 (upstream/main, main)
Author: Marcin Tojek <mtojek@users.noreply.github.com>
Date:   Wed Jan 11 16:05:42 2023 +0100
    docs: API enterprise (#5625)
    * docs: audit, deploymentconfig, files, parameters
    * Swagger comments in workspacebuilds.go
    * structs in workspacebuilds.go
    * workspaceagents: instance identity
    * workspaceagents.go in progress
    * workspaceagents.go in progress
    * Agents
    * workspacebuilds.go
...
```

I know that this can really long.

Reply:
Great question. I haven’t given it much thought to what it _should_ look like. 

Personally I Usually either remove the bullets entirely, rewrite it as proper paragraphs or only keep/rewrite certain relevant bullet points.

Using the type/prefix is not necessary for these IMO. But it’s also not a crime to use them 😄

Reply:
SGTM!